### PR TITLE
fix(DB/gameobjects): Mineral veins on Bloodmyst Isle

### PR DESCRIPTION
**Pull request by Zoidwaffle**
## Changes Proposed:
-  Increase amount of mineral veins from 3 -> 122
- Add them all to a single pool, size 30
- Nodes respawn, randomly, after 5 minutes if taken

## Issues Addressed:
- Based upon data from Gathermate - I tried not to spawn similar objects too close, not spawn things in wrong locations etc..

## Tests Performed:
- Applied SQL
- All locations visited...

## How to Test the Changes:

- Learn the proper tracking skill: `.learn 2580` (mining), `.learn 2383` (herbs), `.learn 2481` (treasure)
- `.tele bloodmyst`
- `.mod speed 10` (or faster)
- `.gm fly on`
- Fly around looking for minerals/herbs/treasure

## Known Issues and TODO List:

- Even though I checked all locations, I cannot guarantee some NPC isn't spawning inside a node. Hopefully we are at worst talking minor adjustments.

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Zoidwaffle** [comment]
I also added a PR for treasures... there is work being done on a PR for herbs as well. So consider testing all 3 PR's at once when they are ready.

File: data/sql/updates/pending_db_world/rev_1614780210274517793.sql [hunk 1]
```diff
@@ -0,0 +1,271 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1614780210274517793');
+
+-- Rewrite positions for minerals for zone: Bloodmyst Isle
+
+-- Before: 3 locations
+-- After: 122 locations
```
Comment (Si1ker) [inline data/sql/updates/pending_db_world/rev_1614780210274517793.sql:6]: Please avoid including stuff into the SQL update that either are already on the PR or "can be" placed in the PR.
The other short comments below to let the lector know about the intention of the query is ok.

Comment (Zoidwaffle) [reply data/sql/updates/pending_db_world/rev_1614780210274517793.sql]: OK, I removed 2 of those lines.

**Review by Si1ker — APPROVED** [review]

**Comment by Zoidwaffle** [comment]
OK, so checked out all locations.

Found like 28 mineral veins (not the expected 30)... Video: https://youtu.be/IwwQLq61_FU (remember you can speed it up).

**Comment by Mitradis** [comment]
if you increased amount of mineral veins from 3 -> 122 will this exacerbate this problem?
https://github.com/azerothcore/azerothcore-wotlk/issues/2510
or is it a separate location and therefore this problem will not affect it

**Comment by Zoidwaffle** [comment]
@Mitradis : I can test it later. I made an "updated" version of my "visit everything" script... So it's busy visiting the 285 herb locations and trying to track the amount automatically without me having to do anything. I can try it out with different pool values to see if some weirdness is going on.

**Comment by Zoidwaffle** [comment]
... and a bit more.

In theory.... If anyone went over all the zones, then everything could get "cleaned up" and this issue would hopefully solve itself.

I have made some simple shell scripts to help out in that regard, but it still takes some effort to fix a zone and there's a lot of those... Of course, not all zones are "broken" - from looking at the database investigating this stuff, I can see that some zones got a lot of love and attention already while others are clearly in need of a do-over.

Bloodmyst just happened to be really bad, so it was my first attempt of a major fix in this regard.

If there are other Linux users around, I could share my simple scripts and perhaps more could work on fixing stuff - because fixing a zone is easily ~2 hours of work, even if it's semi-automated...

**Comment by Zoidwaffle** [comment]
bloodmyst_minerals

23 of 30 mineral veins found, this was with ~30 herbs and ~10 treasure activated in the zone.

**Comment by Zoidwaffle** [comment]
This is super weird... I tried increasing it to 50 spawns, but only 29 appeared (total with herbs. etc. 90).

Then 10 of which 8 appeared (total 50 objects)

Back to 30 of which 25 appeared... (total of 40 objects)

I really don't understand exactly what happens - I can understand I see a few nodes less because of weird angles when teleporting around, but I would expect that to be 10% or less.

As my final test.

Spawn 30 herbs, 30 minerals and 10 chests.... then visit every spot and count...

9/10 treasures.
31/30 herbs (wut?)
25/30 minerals.
A total of 65 our of an expected 70 spawns. It's within my 10% expected error margin at least!

bloodmyst_total

So it seems like the numbers set for the pool are close but not accurate. The weird part is 31 herbs when only 30 should be there... If anyone with understanding of the spawning mechanics have some information regarding this, it would be awesome.

TL;DR: I think this PR is an overall improvement, even if the pool numbers are apparently approximated.

**Comment by Zoidwaffle** [comment]
I redid my experiment... visited 435 nodes (yet, there's that many) once more on a restarted core. So the locations should be different when compared to the one above.

28/30 herbs.
9/10 treasure.
29/30 minerals.

A total of 66 of 70 objects found.
bloodmyst_all2
